source/mqtt3_to_mqtt5_adapter.c

```diff
@@ -174,8 +939,10 @@ struct aws_mqtt_client_connection *aws_mqtt_client_connection_new_from_mqtt5_cli
 
     /*
      * We start disabled to handle the case where someone passes in an mqtt5 client that is already "live."
-     * In that case, we don't want callbacks coming back before construction is even over, so instead we "cork"
-     * things by starting in the disabled state.  We'll enable the adapter as soon as they try to connect.
+     * We'll enable the adapter as soon as they try to connect via the 311 interface.  This
+     * also ties in to how we simulate the 311 implementation's don't-reconnect-if-initial-connect-fails logic.
+     * The 5 client will continue to try and reconnect, but the adapter will go disabled making it seem to the 311
```

Comment:
Question: If it tries to reconnect, can it successfully reconnect? If it does but the MQTT311 user seems to think it is offline, what happens if a publish comes in after reconnecting? Will the callback still be invoked?

If callbacks are not invoked if the client reconnects but the MQTT311 user sees it as offline, then doesn't that mean QoS1 messages will not be queued waiting for re-connection on a resumed session?

For example:
* MQTT5 client with MQTT311 connections and subscribes to `test/topic`.
* MQTT5 client disconnects due to user initiated disconnect.
* MQTT5 client tries to connect due to user initiated connect. `clean_start` is false and is within time to resume the session.
* MQTT5 client with MQTT311 interface becomes disconnected on initial connect request, but then reconnects. MQTT311 user sees it as offline.
* A publish is made to `test/topic`
* ????

I'm not necessarily sure if this is an issue or not, as I'm not 100% positive if subscriptions are resumed upon connecting after disconnecting, but if subscriptions are resumed, then I think there might be a quirk to work around (or at least document) in this case, since the customer might get callbacks invoked when they are "offline" or miss QoS1 publishes that they'd expect to be queued waiting when they are "online" from the perspective of the MQTT311 adapter.

Reply:
That makes sense. In that case, I think as long as we document it somewhere then it's all good. I was just not sure what would happen initially.